### PR TITLE
Create a ``chronos_rerun`` script for rerunning past jobs

### PR DESCRIPTION
**Pull request by Rob-Johnson**

### CODE REVIEW

File: paasta_tools/chronos_rerun.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+# limitations under the License.
+"""
+Chronos Rerun: Designed to 'rerun' jobs with the right parameters.
+
+If the job is a regular schedueled job, then chronos_rereun will use the
+/scheduler/job endpoint to manually start a job now with the arguments set to
+the values they would have been on the date specified.
+
+In the case of a parent job, that is a scheduled job where other jobs have it
+listed as a parent, chronos_cleanup will create a new, temporary clone of the
```
Comment (giuliano108) [inline paasta_tools/chronos_rerun.py]: Did you mean: chronos_*rerun* will create a new, temporary clone

File: paasta_tools/chronos_tools.py [hunk 2]
```diff
@@ -671,6 +664,7 @@ def parse_time_variables(input_string, parse_time=None):
     :param parse_time: Reference Datetime object to parse the date and time strings, defaults to now.
     :returns: A string with the date and time variables replaced
     """
+    print 'parse_time: %s' % parse_time
```
Comment (giuliano108) [inline paasta_tools/chronos_tools.py]: Leftover debugging `print`?

File: paasta_itests/chronos_rerun.feature [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  Scenario: a job can be rerun
+    Given a working paasta cluster
+      And we have yelpsoa-configs for the service "testservice" with enabled scheduled chronos instance "testinstance"
+      And we have a deployments.json for the service "testservice" with enabled chronos instance "testinstance"
+     When we run chronos_rerun for service_instance testservice testinstance
+     Then we should get exit code 0
+     When we store the name of the job for the service testservice and instance testinstance as myjob
+     Then the field "disabled" for the job stored as "myjob" is set to "False"
+
+  Scenario: a dependent job is converted to a scheduled job
```
Comment (solarkennedy) [inline paasta_itests/chronos_rerun.feature]: Can you describe why this is needed?

**Comment by Rob-Johnson** [comment]
unblocked by #354 - this is ready for review

**Comment by Rob-Johnson** [comment]
TODO: check if you already have a rerunning chronos job before submitting another (to avoid nasty side effects of writing over your current job) - later we can add a more unique identifier (add the time to the job name maybe?)

File: paasta_tools/chronos_rerun.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    parser = argparse.ArgumentParser(
+        description='',
+    )
+    parser.add_argument('-v', '--verbose', action='store_true', dest="verbose", default=False,
+                        help="Print out more output regarding the state of the service")
+    parser.add_argument('-d', '--soa-dir', dest="soa_dir", metavar="SOA_DIR",
+                        default=service_configuration_lib.DEFAULT_SOA_DIR,
+                        help="define a different soa config directory")
+    parser.add_argument('service_instance', help='Instance to operate on. Eg: example_service.main')
+    parser.add_argument('execution_date', help="The date the job should be rerun for")
```
Comment (giuliano108) [inline paasta_tools/chronos_rerun.py]: I understand this is the backend part, but should we mention which date format we expect, here? (`%Y-%m-%dT%H:%M:%S`)

File: paasta_tools/chronos_rerun.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    of the schedule is irrelevant, but required by Chronos.
+    """
+    chronos_job['schedule'] = 'R1//PT1M'
+    return chronos_job
+
+
+def set_tmp_naming_scheme(chronos_job):
+    """
+    Given a chronos job, return a new job identical to the first, but with the
+    schedule replaced with one that will set the job to run now.
```
Comment (giuliano108) [inline paasta_tools/chronos_rerun.py]: > Given a chronos job, return a new job identical to the first, but with the schedule replaced with one that will set the job to run now.

Wrong, copy/pasted, docstring?

File: paasta_tools/chronos_rerun.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    timestamp = datetime.datetime.utcnow().isoformat()
+    timestamp = timestamp.replace(':', '')
+    timestamp = timestamp.replace('.', '')
+
+    chronos_job['name'] = '%s-%s%s%s' % (chronos_tools.TMP_JOB_IDENTIFIER,
+                                         timestamp,
+                                         chronos_tools.SPACER,
+                                         current_name)
+
+    print chronos_job['name']
```
Comment (giuliano108) [inline paasta_tools/chronos_rerun.py]: Stray `print`?

File: paasta_tools/chronos_tools.py [hunk 7]
```diff
@@ -129,9 +132,18 @@ def compose_job_id(service, instance):
...
+    decomposed = job_id.split(SPACER)
+    if len(decomposed) == 3:
+        if not decomposed[0].startswith(TMP_JOB_IDENTIFIER):
+            raise InvalidJobNameError('invalid job id %s' % job_id)
+        else:
+            return (decomposed[1], decomposed[2])
+    elif len(decomposed) == 2:
+        return (decomposed[0], decomposed[1])
+    else:
+        raise InvalidJobNameError('invalid job id %s' % job_id)
```
Comment (giuliano108) [inline paasta_tools/chronos_tools.py]: So Chronos jobs can't ever carry a `git_hash` and a `config_hash`, right?

Comment (Rob-Johnson) [inline paasta_tools/chronos_tools.py]: Correct - we removed that a while back

**Comment by giuliano108** [comment]
ShipIt!

**Pull request merged**